### ISSUE TITLE
Exception on line break in small columns

### ISSUE THREAD

**Issue by jkDesignDE** [xhtml2pdf/xhtml2pdf#307]
I have often exceptions when using small text columns. It can be mostly reproduced when using:
- small columns (about 50mm)
- monotype fonts (Courier)
- words ends with '.' or ')'

Most crashes are in functions _getFragWords(frags) and in split(text, delim=None) at file reportlab_paragraph.py.
The reason is a mismatch between str and bytes (text=b'ABC' delim=' ').

I fixed it with some changes (see below) but I don't understand the philosophy behind your code...
Sometimes this functions are called with type str and sometimes with type bytes - maybe the bug is in the calling function.

```
def split(text, delim=None):
    if type(text) is bytes:
        try:
            text = text.decode('utf-8')
        except Exception:
            pass
    if type(delim) is bytes:
        try:
            delim = delim.decode('utf-8')
        except Exception:
            pass
    elif delim is None and '\xa0' in text:
        return [uword.encode('utf-8') for uword in _wsc_re_split(text)]
    return [uword.encode('utf-8') for uword in text.split(delim)]
```
```
def _getFragWords(frags):
    ...
    for f in frags:
        text = f.text
        if type(text) is bytes:   #JK
            try:
                text = text.decode('utf-8')
            except Exception:
                text = '???'
        # of paragraphs
        if text != '':
    ...
```

**Comment by benjaoming** [comment]
Hi @jkDesignDE 

Thanks for looking into this. Are you able to make a PR with your changes?

We are currently working in the `develop` branch where 0.1 will be released from. Many changes have been made, also in regards to:

> I don't understand the philosophy behind your code...
Sometimes this functions are called with type str and sometimes with type bytes

Some of the confusion is probably caused by Python2+3 friendly code. And I can't say with certainty that it's correct. But tests seem to pass, and we have more and more coverage.

Also notice that exceptions should be allowed to happen, catching them is one of the other absent "philosophies" of the codebase - see #290 

If stuff is horribly broken for you, have a look at [weasyPrint](https://github.com/Kozea/WeasyPrint), another project that will be announced as a better option once xhtml2pdf 0.1 is out and this project goes to Valhalla.

**Comment by jkDesignDE** [comment]
Thx for the fast reply.
I now forked the 'develop' branch and found 2 more places that crashes. I'm just looking for more... PR will come soon.

But with the 'develop' branch I get exceptions on loading fonts (@font-face {... as well as <link href='https://fonts.googleapis.com/...). Are you still working on it??? Next issue???

**Comment by benjaoming** [comment]
Yes, I think you are correct that loading resources from remote URLs might be broken, you are welcome to work out the causes and open a separate issue for that :) Perhaps this is a good case for doing some test-driven development :)

I think most people load fonts from a local absolute path (`/path/to/font.ttf`), maybe that's why the issue has gone unnoticed.
